distest-test-bot.py

```diff
@@ -26,13 +26,46 @@
 async def test_reverse(interface):
     await interface.assert_reply_contains("!reverse this class sucks", "skcus ssalc siht")
 
+@test()
+async def test_joke(interface):
+    # Random response is hard to test; ? is the only thing common to all
+    # responses
+    await interface.assert_reply_contains("!joke", "?")
```

Comment:
So, is this just testing that the question has a question mark in it? I can definitely see how a random response would be a bit tricky to test.  Is there a way to have it check that what was outputted is equal to one of the elements in the list, possibly?

Reply:
Essentially, yes. In theory I could test that it matches something in the list, but as the list is a local variable in the command function, changes to the command function would be needed in order to do so (the list needs to be a module global variable in order for external code to inspect it). I did not want to perform any changes to the actual code beyond the minimum necessary.

Reply:
Ahh, that makes sense. Yeah, this is plenty good for what we need 😁